gateway/ln-gateway/src/utils.rs

```diff
@@ -0,0 +1,36 @@
+use std::{future::Future, result::Result, thread::sleep, time::Duration};
+
+use tracing::info;
+
+/// Retry an operation util the operation succeeds, OR
+/// The maximum number of attempts are made without success
+pub async fn retry<F, R, T>(
+    op_name: String,
+    op_fn: F,
+    wait: Duration,
+    max_retries: u32,
+) -> Result<T, anyhow::Error>
+where
+    F: Fn() -> R,
+    R: Future<Output = Result<T, anyhow::Error>>,
+{
+    for _ in 0..max_retries {
+        match op_fn().await {
+            Ok(result) => return Ok(result),
+            Err(e) => {
+                info!(
+                    "{} failed with error: {}. Retrying in {} seconds",
+                    op_name,
+                    e,
+                    wait.as_secs()
+                );
+                sleep(wait);
+            }
+        }
+    }
+
+    Err(anyhow::anyhow!(format!(
```

Comment:
I would consider returning the original error instead `if max_retries < retries {`